lms/djangoapps/discussion_api/serializers.py

```diff
@@ -191,12 +191,12 @@ class ThreadSerializer(_ContentSerializer):
     closed = serializers.BooleanField(read_only=True)
     following = serializers.SerializerMethodField()
     comment_count = serializers.SerializerMethodField(read_only=True)
```

Comment:
Again - thank you for clearing up the code change a bit.  However, I'm unclear what the current defect is and whether this is the right fix.

We have 3 possible states:
1. Thread was never read.
2. Thread was read at least once, but there are no unread comments.
3. Thread was read at least once, and there are some unread comments.

It seems like the deleted serializer `get_unread_comment_count` assumed that `read` status is false for "never read" (1 above), and true for "ever read" (2 and 3 above).  

If a thread was never read, it makes sense to me that the unread_comment_count should match the comment_count, especially if these are displayed near each other.

Maybe you can provide more details about what defect is being fixed, and whether it is related to how Mobile is displaying this data?